backend/Repositories/UserRepo.cs

```diff
@@ -20,7 +20,7 @@ public UserRepo(WhaleSpottingDbContext context)
 
         public User GetByUsername(string username)
         {
-            return _context.Users.Single(user => user.Username == username);
+            return _context.Users.SingleOrDefault(user => user.Username == username);
```

Comment:
Rather than this coupled with a null check further up the chain, prefer keeping `Single()` here and explicitly try/catch-ing the errors up in the service